### PR TITLE
AWS S3 repositories using `credential_process` to authenticate

### PR DESCRIPTION
**Pull request by t0rr3sp3dr0**
AWS allows users to configure a profile to retrieve credentials using a third-party binary by setting `credential_process` attribute in `~/.aws/config`. This is commonly used by SSO solutions to provide credentials to applications using AWS SDK.

Older versions of AWS SDK for Java, such as the one currently shipped in Gradle, do not support this attribute. When trying to use a AWS profile with `credential_process` to fetch an S3 bucket used as a dependency repository, it fails due to lack of support of AWS SDK currently bundled in Gradle.

Since version `1.11.489`, AWS SDK for Java supports this attribute, but Gradle is using version `1.11.407`. https://github.com/aws/aws-sdk-java/blob/master/CHANGELOG.md#111489-2019-01-24

I've upgraded AWS SDK to the latest version available to take advantage of this feature and possibly many others. It was necessary to upgrade `org.apache.httpcomponents:httpclient` and
`org.apache.httpcomponents:httpcore` for it to work with this new version of AWS SDK.

### CODE REVIEW

**Comment by t0rr3sp3dr0** [comment]
@big-guy, should I worry about these errors on TeamCity? I'm not able to see them.

image

**Comment by big-guy** [comment]
@t0rr3sp3dr0 did you login as a guest? You should be able to see it that way.

**Comment by t0rr3sp3dr0** [comment]
Actually I was logging in using by personal account. It worked fine when I used the guest option. Thanks!

**Comment by t0rr3sp3dr0** [comment]
@big-guy I’ve fixed the problem on TeamCity

**Comment by t0rr3sp3dr0** [comment]
Sorry, I've missed `jmespath-java` but now it should be ok!

File: gradle/dependencies.gradle [hunk 1]
```diff
@@ -37,15 +37,15 @@ libraries.asm_commons =         [coordinates: 'org.ow2.asm:asm-commons', version
 libraries.asm_tree =            [coordinates: 'org.ow2.asm:asm-tree', version: libraries.asm.version, license: "3-Clause BSD"]
 libraries.asm_util =            [coordinates: 'org.ow2.asm:asm-util', version: libraries.asm.version, license: "3-Clause BSD"]
 libraries.asm_analysis =        [coordinates: 'org.ow2.asm:asm-analysis', version: libraries.asm.version, license: "3-Clause BSD"]
-libraries.awsS3_core =          [coordinates: 'com.amazonaws:aws-java-sdk-core', version: '1.11.407', license: "Apache 2.0"]
+libraries.awsS3_core =          [coordinates: 'com.amazonaws:aws-java-sdk-core', version: '1.11.627', license: "Apache 2.0"]
```
Comment (ljacomet) [inline gradle/dependencies.gradle]: `1.11.633` is available now, could you update your PR with this version?

**Comment by t0rr3sp3dr0** [comment]
@ljacomet I've updated AWS SDK to version 1.11.633 as requested!

File: subprojects/distributions/src/integTest/groovy/org/gradle/DistributionIntegritySpec.groovy [hunk 2]
```diff
@@ -115,8 +115,8 @@ class DistributionIntegritySpec extends DistributionIntegrationSpec {
             'plugins/google-oauth-client-1.25.0.jar' : '7e2929133d4231e702b5956a7e5dc8347a352acc1e97082b40c3585b81cd3501',
             'plugins/gson-2.8.5.jar' : '233a0149fc365c9f6edbd683cfe266b19bdc773be98eabdaf6b3c924b48e7d81',
             'plugins/hamcrest-core-1.3.jar' : '66fdef91e9739348df7a096aa384a5685f4e875584cce89386a7a47251c4d8e9',
-            'plugins/httpclient-4.5.6.jar' : 'c03f813195e7a80e3608d0ddd8da80b21696a4c92a6a2298865bf149071551c7',
-            'plugins/httpcore-4.4.10.jar' : '78ba1096561957db1b55200a159b648876430342d15d461277e62360da19f6fd',
+            'plugins/httpclient-4.5.9.jar' : '38b9f16f504928e4db736a433b9cd10968d9ec8d6f5d0e61a64889a689172134',
```
Comment (ljacomet) [inline subprojects/distributions/src/integTest/groovy/org/gradle/DistributionIntegritySpec.groovy]: This needs to be bumped as well to `4.5.10`

**Comment by t0rr3sp3dr0** [comment]
@ljacomet my bad. Now it's fixed!

**Comment by t0rr3sp3dr0** [comment]
Can I trigger TeamCity tests somehow? For what I've seen it needs to be triggered manually by someone.

**Comment by ljacomet** [comment]
Thanks for the quick follow up!

No you cannot trigger a TC build, we have Travis for minimal checks whenever a PR is pushed. For a full "Ready For Merge" run, it needs to be triggered by a Gradle engineer.

Just did 🤞

**Comment by t0rr3sp3dr0** [comment]
@ljacomet one test failed due to a network problem. Should I worry about that?

```
Could not HEAD 'https://oss.sonatype.org/content/repositories/snapshots/org/openjdk/jmc/common/7.0.0-SNAPSHOT/common-7.0.0-20190108.115032-20.jar'. Received status code 502 from server: Bad Gateway
```

**Comment by ljacomet** [comment]
No that's most likely a transient error. I triggered a rebuild to be sure.

**Pull request merged**